service/controller/v13/resource/instance/create.go

```diff
@@ -4,21 +4,30 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/giantswarm/azure-operator/service/controller/v13/key"
-	"github.com/giantswarm/azure-operator/service/controller/v13/resource/instance/internal/state"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+
+	"github.com/giantswarm/azure-operator/service/controller/v13/key"
+	"github.com/giantswarm/azure-operator/service/controller/v13/resource/instance/internal/state"
 )
 
 // configureStateMachine configures and returns state machine that is driven by
 // EnsureCreated.
 func (r *Resource) configureStateMachine() {
 	sm := state.Machine{
-		DeploymentUninitialized: r.deploymentUninitializedTransition,
-		DeploymentInitialized:   r.deploymentInitializedTransition,
-		ProvisioningSuccessful:  r.provisioningSuccessfulTransition,
-		InstancesUpgrading:      r.instancesUpgradingTransition,
-		DeploymentCompleted:     r.deploymentCompletedTransition,
+		DeploymentUninitialized:     r.deploymentUninitializedTransition,
+		DeploymentInitialized:       r.deploymentInitializedTransition,
+		ProvisioningSuccessful:      r.provisioningSuccessfulTransition,
+		MasterInstancesUpgrading:    r.masterInstancesUpgradingTransition,
+		WaitForMastersToBecomeReady: r.waitForMastersToBecomeReadyTransition,
+		ScaleUpWorkerVMSS:           r.scaleUpWorkerVMSSTransition,
+		CordonOldWorkers:            r.cordonOldWorkersTransition,
+		WaitForWorkersToBecomeReady: r.waitForWorkersToBecomeReadyTransition,
+		CreateWorkerDrainerConfigs:  r.createWorkerDrainerConfigsTransition,
+		DeleteWorkerDrainerConfigs:  r.deleteWorkerDrainerConfigsTransition,
+		TerminateOldWorkerInstances: r.terminateOldWorkersTransition,
+		ScaleDownWorkerVMSS:         r.scaleDownWorkerVMSSTransition,
+		DeploymentCompleted:         r.deploymentCompletedTransition,
```

Comment:
I just thought you can replicate the same with the operatorkit resources and the controller context. I see the need and the complexity but I think it is not a good practice to manage all of that logic in one resource package. We are pretty much replicating the reconciliation primitives inside the reconciliation primitives somehow. Just food for thought.

Reply:
There could be one dedicated controller for this. Then it would be cohesive as well.

Reply:
Do you have idea how that could synchronize with the "main" one if they reconcile the same CR? Specific condition flag in status?

Reply:
I am not on top of all the internals and requirements. Can't make a good suggestion at this point.